core/src/main/java/org/apache/iceberg/rest/TableScanResponseContext.java

```diff
@@ -0,0 +1,40 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+package org.apache.iceberg.rest;
+
+import java.util.Map;
+import org.apache.iceberg.PartitionSpec;
+
+public class TableScanResponseContext {
```

Comment:
Does this need to be public? I feel like this should just be nested inside `RESTSerializers`. The parser APIs can just take the actual values instead of the context variable, which I feel like is a better API (I don't want to have to build a context neccessarily just to parse things if I have the specs from somewhere else)

Reply:
Right, I think this should just be an intermediate object used inside the parser. but the fromJson takes in separate parameters for specsById/case-sensitive.